crates/ruff_python_formatter/src/context.rs

```diff
@@ -210,3 +220,109 @@ where
             .set_node_level(self.saved_level);
     }
 }
+
+/// The current indent level of the formatter.
+///
+/// One can determine the the width of the indent itself (in number of ASCII
+/// space characters) by multiplying the indent level by the configured indent
+/// width.
+///
+/// This is specifically used inside the docstring code formatter for
+/// implementing its "dynamic" line width mode. Namely, in the nested call to
+/// the formatter, when "dynamic" mode is enabled, the line width is set to
+/// `min(1, line_width - indent_level * indent_width)`, where `line_width` in
+/// this context is the global line width setting.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub(crate) struct IndentLevel {
+    /// The numeric level.
+    level: usize,
+}
+
+impl IndentLevel {
+    /// Returns a new indent level for the given value.
+    pub(crate) fn new(level: usize) -> IndentLevel {
+        IndentLevel { level }
+    }
+
+    /// Returns the next indent level.
+    pub(crate) fn next(self) -> IndentLevel {
```

Comment:
Nit: To me the semantic of `next` for an indent level isn't immediately clear. I would find`increment` and `decrement` easier to understand.